### ISSUE TITLE
ArgumentError while running SpaceAroundOperators

### ISSUE THREAD

**Issue by digitalcora** [rrrene/credo#731]
### Environment

* Credo version (`mix credo -v`): **1.1.5** (also tested on 1.2.0-rc2)
* Erlang/Elixir version (`elixir -v`): **1.8.1**
* Operating system: **OS X 10.15**

### Description

When I run Credo on my project using the default config, it crashes with the following error:

```
** (ArgumentError) cannot convert the given list to a string.

To be converted to a string, a list must contain only:

  * strings
  * integers representing Unicode codepoints
  * or a list containing one of these three elements

Please check the given list or call inspect/1 to get the list representation, got:

[{{67, 18, 67}, [{:identifier, {67, 20, nil}, :date_type}]}, "_dates"]

    (elixir) lib/list.ex:888: List.to_string/1
    lib/credo/code/token.ex:96: Credo.Code.Token.PositionHelper.position_tuple/3
    lib/credo/check/consistency/space_around_operators/space_helper.ex:51: Credo.Check.Consistency.SpaceAroundOperators.SpaceHelper.no_space_between?/2
    lib/credo/check/consistency/space_around_operators/collector.ex:164: Credo.Check.Consistency.SpaceAroundOperators.Collector.without_space?/3
    lib/credo/check/consistency/space_around_operators/collector.ex:128: Credo.Check.Consistency.SpaceAroundOperators.Collector.record_spaces/4
    lib/credo/check/consistency/space_around_operators/collector.ex:35: Credo.Check.Consistency.SpaceAroundOperators.Collector.traverse_tokens/3
    lib/credo/check/consistency/collector.ex:140: anonymous fn/3 in Credo.Check.Consistency.Collector.find_issues/4
    (elixir) lib/enum.ex:1327: Enum."-map/2-lists^map/1-0-"/2
Function: #Function<10.15291192/0 in Credo.Check.Runner.run/2>
    Args: []
```

From the available context, it seems to be failing to parse the following code:

```elixir
  defp dates(service, attributes) do
    service =
      Enum.reduce(["added", "removed"], service, fn date_type, acc ->
        %{acc | "#{date_type}_dates": Map.get(attributes, date_type <> "_dates", [])}
      end)

    Enum.reduce(["start", "end"], service, fn date_type, acc ->
      date =
        case Map.get(attributes, date_type <> "_date") do
          nil -> nil
          date_string -> Date.from_iso8601!(date_string)
        end

      %{acc | "#{date_type}_date": date}
    end)
  end
```

Adding `# credo:disable-for-this-file` to the file doesn't fix the crash, even though this file is the only place the token `date_type` appears in the codebase. The only thing that seems to work is disabling the `SpaceAroundOperators` check entirely.

### Related Issues

This kind of error is also seen in #494, #556, #681, and #701. However, it appears to be a new instance of the error — the most recent one was reported as fixed with version 1.1.5, but I am using that version and still get a crash. (I also tried with 1.2.0-rc2 but that did not fix it)

**Comment by rrrene** [comment]
Thx for reporting! :+1:

**Comment by TheFirstAvenger** [comment]
I was able to reproduce this on the latest master and Elixir 1.9.4. This minimal test fails:
```elixir
  test "it should not crash for issue #731" do
    [
      ~S"""
      %{acc | "#{date_type}_dates": :foo}
      """
    ]
    |> to_source_files()
    |> refute_issues(@described_check)
  end
```

On this line:
https://github.com/rrrene/credo/blob/master/lib/credo/check/consistency/space_around_operators/space_helper.ex#L54
arg2 is:
```elixir
{:kw_identifier_unsafe, {1, 9, nil},
 [
   {{1, 10, nil}, {1, 21, nil}, [{:identifier, {1, 12, nil}, :date_type}]},
   "_dates"
 ]}
```
Which when passed to Token.position fails on this line:
https://github.com/rrrene/credo/blob/master/lib/credo/code/token.ex#L102

I haven't fully grokked what is supposed to be happening, but my hunch is something earlier in the path should recognize the `:kw_identifier_unsafe` tuple and handle it different?

**Comment by rrrene** [comment]
@TheFirstAvenger :+1: Thx for the analysis!

@digitalcora This highlighted a deeper issue with interpolated keys in maps not being handled correctly. Thx so much for reporting! 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by digitalcora** [comment]
@rrrene Thanks! Your fix has fixed the crash in our project 😊

**Comment by rrrene** [comment]
Thx for reporting this! :+1: It should be be fixed in the latest RC of Credo (`v1.2.0-rc4`).

If it is not, please feel free to re-open this issue!
